api/internal/localizer/locloader.go

```diff
@@ -0,0 +1,382 @@
+// Copyright 2022 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package localizer
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"path/filepath"
+	"strings"
+
+	"sigs.k8s.io/kustomize/api/ifc"
+	"sigs.k8s.io/kustomize/api/internal/git"
+	"sigs.k8s.io/kustomize/api/loader"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+const dstPrefix = "localized"
+const localizeDir = "localized-files"
+const fatalPrefix = "should never happen"
+
+const errAlpha = "alpha unsupported localize feature"
+const errCleanDst = "unable to clean localize destination"
+const errReference = "invalid localize reference location '%s'"
+
+var errDstRef = errors.Errorf("reference into localize destination")
+var errRootOutsideScope = errors.Errorf("local root not within localize scope")
+
+// locArgs stores localize arguments
+type locArgs struct {
+
+	// directory that bounds target's local references, empty string if target is remote
+	scope filesys.ConfirmedDir
+
+	// localize destination
+	newDir filesys.ConfirmedDir
+}
+
+// LocLoader is the Loader for kustomize localize.
+//
+// Each LocLoader is created and Loads under localize constraints. For each input path, LocLoader
+// also outputs the corresponding path to the localized copy.
+type LocLoader struct {
+	fSys filesys.FileSystem
+
+	args *locArgs
```

Comment:
nit: use two fields "scope" and "newDir" directly in LocLoader instead of combining them into another struct.

Reply:
As discussed, I did this to save memory, avoid copying, and group the arguments together, but I also see your point that these aren't strong reasons. Fixed.